### PR TITLE
Cropping poc

### PR DESCRIPTION
**Pull request by jarodabel**
Sorry about the formatting changes. 

I made these changes to allow the point cloud to be "cropped". I am looking for any feedback on this approach. 

Here are some more details on my use case and solution 
https://github.com/mkkellogg/GaussianSplats3D/discussions/306#discussioncomment-10537007

### CODE REVIEW

File: src/splatmesh/SplatMaterial3D.js [hunk 1]
```diff
@@ -60,6 +60,15 @@ export class SplatMaterial3D {
             'type': 'i',
             'value': 0
         };
+        // Add the crop box uniforms with a default size of 2x2x2
```
Comment (jarodabel) [inline src/splatmesh/SplatMaterial3D.js:63]: establish the BBox corners

File: src/splatmesh/SplatMaterial3D.js [hunk 2]
```diff
@@ -224,14 +232,26 @@ export class SplatMaterial3D {
...
             varying vec4 vColor;
             varying vec2 vUv;
             varying vec2 vPosition;
+            varying vec3 vWorldPosition;  // Use the world position passed from the vertex shader
+
         `;
 
         fragmentShaderSource += `
             void main () {
+                // Crop box logic: discard fragments outside the crop box
```
Comment (jarodabel) [inline src/splatmesh/SplatMaterial3D.js:247]: use the BBox corners

**Comment by mkkellogg** [comment]
I like how simple this solution is, but I'm hesitant to add it right now because a more general solution might be better (e.g. one that supports multiple bounding boxes, different shape-based bounding regions (spheres, etc...), or possibly even more complex bounding regions based on functions or meshes).

Does your use-case for this feature require the fraction of splats that are inside the bounding box to be rendered while the remaining portion is not? If an "all-or-nothing" approach for each splat is acceptable, then moving the bounding box test to the vertex shader would be more performant.

Additionally I think it would be better to "opt-in" to this feature so that the bounding box test is not always run, regardless of whether or not the user is using it (again, for the sake of performance).

**Comment by jarodabel** [comment]
Thanks for the tip on using the vertex shader. 

I think all or nothing is working for me. 

Agree on opt-in.

**Pull request closed without merging**